tests/eval_files/async102.py

```diff
@@ -111,7 +111,9 @@ async def foo():
         myvar = True
         with trio.open_nursery(10) as s:
             s.shield = myvar
-            await foo()  # safe in theory, error: 12, Statement("try/finally", lineno-6)
+            # this is not safe in theory - because `trio.open_nursery` is an async cm,
+            # so it's not possible to open a nursery at all.
+            await foo()  # error: 12, Statement("try/finally", lineno-8)
```

Comment:
So this should be `async with`, yeah?  Also nurseries only checkpoint on exit, so I think it _is_ safe?

Reply:
wait, it doesn't block/checkpoint on exit either??

```python
import trio

async def foo():
    with trio.CancelScope() as cs:
        cs.cancel()
        async with trio.open_nursery():
            print("in nursery")
        print("after nursery")

trio.run(foo)
```

```sh
python foo.py
in nursery
after nursery
```

Reply:
> It does not block on entry; on exit it blocks until all child tasks have exited.

oh, hm. I guess that is meant to be read "if there are child tasks still alive, block until they have exited. If there are no child tasks, or they have all exited, it doesn't block."
Which in the end means that `open_nursery` never checkpoints on entry, and *sometimes* checkpoints on exit.